### ISSUE TITLE
Automatically add absolute flag for external redirects

### ISSUE THREAD

**Issue by qzminski** [contao/contao#2254]
When you create a new page of external redirect type, choosing a page in picker generates a `{{link_url::ID}}` insert tag. This is wrong as then the generated URL is relative and the redirect will point in fact to a 404 page (i.e. visitor is taken to `https://domain.tld/$redirectPageAlias/$redirectUrlTarget`). This is easily fixed by making the URL absolute `{{link_url::ID|absolute}}`, yet regular users tend to forget to add that. Could this flag be added automatically somehow?

Maybe it would be possible with some `onload_callback` or field's `load_callback` in combination with https://github.com/contao/contao/pull/450? One concern is that not all picker providers or insert tags support `|absolute` flag.

Tested on Contao 4.9.5.

/cc @Toflar

**Comment by fritzmg** [comment]
> This is wrong as then the generated URL is relative and the redirect will point in fact to a 404 page

Why does it point to a 404 page? Any relative URL is prepended by the `<base href>` by the browser.

**Comment by leofeyer** [comment]
Is this the same problem as in #2171?

**Comment by qzminski** [comment]
@fritzmg I forgot to mention, it happens when you access the page URL directly. Then the redirect response being thrown/returned contains an invalid URL.

**Comment by fritzmg** [comment]
I see, makes sense now :)

**Comment by ausi** [comment]
I think instead of the `|absolute` flag we should resolve all relative URLs (after replacing the insert tag) so that manually entered relative URLs (and other/custom insert tags) also get absolutized.

**Comment by bennyborn** [comment]
Haven't been able to reproduce this in the online demo. Using the Inser-Tag `{{link_url::xx}}` as well as opening the page directly leads to the desired url...

**Comment by leofeyer** [comment]
The online demo is still at version 4.9.5. 🙈

**Comment by bennyborn** [comment]
> The online demo is still at version 4.9.5. 🙈

So? @qzminski  said he tested it on 4.9.5 ;)

**Comment by leofeyer** [comment]
You are right. My bad.

**Comment by aschempp** [comment]
I generally wonder why we should not always use absolute URLs? I can't imagine it to ever harm and it would solve millions of issues (including https://github.com/contao/contao/pull/2262) 🤷

**Comment by leofeyer** [comment]
> always use absolute URLs

I am for it. @contao/developers Any objections?

**Comment by Toflar** [comment]
That would mean removing the flag and just always generate absolute URL's intstead then? I actually never understood why there are relative ones in the first place either but 🤷‍♂️

**Comment by fritzmg** [comment]
Are we talking about `UrlGeneratorInterface::ABSOLUTE_URL` or `UrlGeneratorInterface::ABSOLUTE_PATH`?

**Comment by ausi** [comment]
> > always use absolute URLs
> 
> I am for it. @contao/developers Any objections?

For `{{link_url}}` only? That would then generate different type of URLs than the navigation module.

If we want to change how URLs work in Contao, we should IMO develop a consistent concept (and drop `<base>` for example) and change all URLs togehter (where possible) in a minor release.

**Comment by aschempp** [comment]
> @fritzmg I forgot to mention, it happens when you access the page URL directly. Then the redirect response being thrown/returned contains an invalid URL.

Can you explain that? You mean if the URL is accessed from the back end? This should be solved by https://github.com/contao/contao/pull/2262 as well.

**Comment by fritzmg** [comment]
> Can you explain that?

I think he means accessing the external redirect page with its alias in the front end, which will generate a redirect response to the external URL. But yes, this is also fixed by #2279.

**Comment by qzminski** [comment]
> Can you explain that? You mean if the URL is accessed from the back end? This should be solved by #2262 as well.

Yes, either via front end preview or the page URL directly. Could be solved in 4.10 but the "issue" remain in 4.9. Steps to reproduce on demo:

1. Create a page of "external redirect" type.
2. Name it "Redirect test" for instance.
3. Launch page picker and pick "About Contao" page with ID 50.
4. Tests:
4a. Open the page directly `https://demo.contao.org/en/redirect-test.html` and you are taken to `https://demo.contao.org/en/en/about-contao.html` (which also seems to be a bit different than my problem).
4b. Open the page via frontend preview and you are taken to `https://demo.contao.org/preview.php/en/preview.php/en/about-contao.html`.

On the first system I have noticed the problem it actually also included the redirect page alias in the end URL (e.g. `https://demo.contao.org/en/redirect-test/en/about-contao.html`), which I cannot exactly reproduce on demo. Maybe it was due to folder URLs setting or something.
